src/execution_plan/execution_plan_modify.c

```diff
@@ -74,14 +90,8 @@ void ExecutionPlan_PushBelow(OpBase *a, OpBase *b) {
 		return;
 	}
 
-	/* Replace A's former parent. */
-	OpBase *a_former_parent = a->parent;
-
-	/* Disconnect A from its former parent. */
-	_OpBase_RemoveChild(a_former_parent, a);
-
-	/* Add A's former parent as parent of B. */
-	_OpBase_AddChild(a_former_parent, b);
+	/* Disconnect A from its parent and replace it with B. */
+	_OpBase_ReplaceOp(a, b);
```

Comment:
I think it would be much clearer if `_OpBase_ReplaceOp` would become:
`_ExecutionPlan_ReplaceOp(parent, old_child, new_child)`
Also the function (_ExecutionPlan_ReplaceOp) shouldn't connect `old_child` as a child of `new_child`
as its name doesn't implies that.